src/fs/dir.rs

```diff
@@ -39,21 +36,27 @@ impl Dir {
         info!("Reading directory {:?}", &path);
 
         let contents = fs::read_dir(&path)?
-                          .map(|result| result.map(|entry| entry.path()))
-                          .collect::<Result<_, _>>()?;
+            .map(|result| result.map(|entry| entry.path()))
+            .collect::<Result<_, _>>()?;
 
         info!("Read directory success {:?}", &path);
         Ok(Self { contents, path })
     }
 
     /// Produce an iterator of IO results of trying to read all the files in
     /// this directory.
-    pub fn files<'dir, 'ig>(&'dir self, dots: DotFilter, git: Option<&'ig GitCache>, git_ignoring: bool, deref_links: bool) -> Files<'dir, 'ig> {
+    pub fn files<'dir, 'ig>(
+        &'dir self,
+        dots: DotFilter,
+        git: Option<&'ig GitCache>,
+        git_ignoring: bool,
+        deref_links: bool,
+    ) -> Files<'dir, 'ig> {
         Files {
-            inner:     self.contents.iter(),
-            dir:       self,
-            dotfiles:  dots.shows_dotfiles(),
-            dots:      dots.dots(),
+            inner: self.contents.iter(),
+            dir: self,
+            dotfiles: dots.shows_dotfiles(),
+            dots: dots.dots(),
```

Comment:
I think that the above is preferred, but it's an unavoidable loss. Eventually we should use rustfmt for the reasons mentioned in the PR.

Reply:
I mean we can maybe set this in `.rustfmt.toml` so it ignores such things ... I agree I looks nicer ... but since using rustfmt I'm actually a lot less opinionated about such things as it just enforces sane defaults for formatting

Reply:
Yea same, and also for projects where I work with other people I'd rather have clear guidelines than perfect code.

Reply:
from quickly skimming the rustfmt configuration options, I couldn't find any way to force this indentation ... we could maybe use `#[rustmt::skip...]` but I'm not sure it's really worth it.

Reply:
Personally, I think slightly worse looking is worth more readable diffs (= easier code reviews)

Reply:
For large tables, like filetype mapping or icon mapping, we should turn off formatting with `#[rustfmt::skip]`.

Reply:
Rustfmt is a must. i do agree with @cfxegbert .. `#[rustfmt:skip]` for large tables, where any edits will be carefully reviewed for formatting

Reply:
I'm gonna skim the code for such places and try to address those with skips.

Reply:
Alright, so I added skips where it seemed appropriate. Since attributes on expressions are still experimental however, there is a bit of a trade-off. There are quite a few functions evaluating to a `match` without explicit `return` statement. In those cases there are two options:
1. just let rustfmt indent it how it wants it.
2. make a statement of it with `return`:
```rust
    pub fn format(self, time: &DateTime<FixedOffset>) -> String {
        #[rustfmt::skip]
        return match self {
            Self::DefaultFormat  => default(time),
            Self::ISOFormat      => iso(time),
            Self::LongISO        => long(time),
            Self::FullISO        => full(time),
            Self::Relative       => relative(time),
        };
    }
```

I went with 1 when the indentation doesn't improve readability a lot, like in two option matches, and with 2 for larger constructs where the indentation actually helps quite a bit. The example above might have actually been more one for option 1.

Opinions?

Reply:
I think outside of the large tables, and `flags.rs`... it might be easy to overthink it. I think just letting RustFmt do its thing, is perfectly fine :+1:

Reply:
Well I’d say most places where there’s `#[rustfmt::skip]` could have done without it, but if you believe it really is better, than let’s go with that (anyway that’s only a fraction of the code…).

Reply:
Yeah, I started of applying them more happily in the beginning and later tried to avoid it if possible so not all might be necessary. At least we are not losing anything and can still remove some of them later on with no trouble.